File: src/main/java/employeetech/com/employee/manager/service/EmployeeService.java
package employeetech.com.employee.manager.service;

import employeetech.com.employee.manager.model.Employee;
import employeetech.com.employee.manager.repo.EmployeeRepo;
import org.springframework.beans.BeanUtils;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Service
@Transactional
public class EmployeeService {
    private final EmployeeRepo employeeRepo;

    public EmployeeService(EmployeeRepo employeeRepo) {
        this.employeeRepo = employeeRepo;
    }

    public Employee addEmployee(Employee employee) {
        Employee newEmployee = new Employee();
        newEmployee.setId(employee.getId());
        newEmployee.setName(employee.getName());
        newEmployee.setEmail(employee.getEmail());
        newEmployee.setTitle(employee.getTitle());
        newEmployee.setPhone(employee.getPhone());
        newEmployee.setImageUrl(employee.getImageUrl());
        newEmployee.setDepartment(employee.getDepartment());
        newEmployee.setEmployeeCode(UUID.randomUUID().toString());
        return employeeRepo.save(newEmployee);
    }

    public List<Employee> findAllEmployees() {
        return employeeRepo.findAll();
    }

    public Employee updateEmployee(Long id, Employee employeeDetails) {
        Optional<Employee> optionalEmployee = employeeRepo.findById(id);
        if (optionalEmployee.isPresent()) {
            Employee employee = optionalEmployee.get();
            BeanUtils.copyProperties(employeeDetails, employee, "id", "employeeCode");
            return employeeRepo.save(employee);
        }
        return null;
    }

    public Employee findEmployeeById(Long id) {
        return employeeRepo.findById(id).orElse(null);
    }

    public void deleteEmployee(Long id) {
        employeeRepo.deleteById(id);
    }
}

File: src/main/kotlin/employeetech/com/employee/manager/service/EmployeeService.kt
package employeetech.com.employee.manager.service

import employeetech.com.employee.manager.model.Employee
import employeetech.com.employee.manager.repo.EmployeeRepo
import org.springframework.beans.BeanUtils
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.UUID

@Service
@Transactional
class EmployeeService(private val employeeRepo: EmployeeRepo) {

    fun addEmployee(employee: Employee): Employee {
        val newEmployee = Employee(
            id = employee.id,
            name = employee.name,
            email = employee.email,
            title = employee.title,
            phone = employee.phone,
            imageUrl = employee.imageUrl,
            department = employee.department,
            employeeCode = UUID.randomUUID().toString()
        )
        return employeeRepo.save(newEmployee)
    }

    fun findAllEmployees(): List<Employee> {
        return employeeRepo.findAll()
    }

    fun updateEmployee(id: Long, employeeDetails: Employee): Employee? {
        return employeeRepo.findById(id).map { employee ->
            BeanUtils.copyProperties(employeeDetails, employee, "id", "employeeCode")
            employeeRepo.save(employee)
        }.orElse(null)
    }

    fun findEmployeeById(id: Long): Employee? {
        return employeeRepo.findById(id).orElse(null)
    }

    fun deleteEmployee(id: Long) {
        employeeRepo.deleteById(id)
    }
}
